File: riistaandroid/src/main/java/fi/riista/mobile/models/shootingTest/ShootingTestParticipant.java
package fi.riista.mobile.models.shootingTest;

import androidx.annotation.Nullable;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class ShootingTestParticipant implements Serializable {
    @JsonProperty("id")
    public long id;

    @JsonProperty("rev")
    public int rev;

    @JsonProperty("firstName")
    public String firstName;

    @JsonProperty("lastName")
    public String lastName;

    @JsonProperty("hunterNumber")
    public String hunterNumber;

    @JsonProperty("mooseTestIntended")
    public boolean mooseTestIntended;

    @JsonProperty("bearTestIntended")
    public boolean bearTestIntended;

    @JsonProperty("deerTestIntended")
    public boolean deerTestIntended;

    @JsonProperty("bowTestIntended")
    public boolean bowTestIntended;

    @JsonProperty("attempts")
    public List<ShootingTestAttemptSummary> attempts = new ArrayList<>();

    @JsonProperty("totalDueAmount")
    public int totalDueAmount;

    @JsonProperty("paidAmount")
    public int paidAmount;

    @JsonProperty("remainingAmount")
    public int remainingAmount;

    @JsonProperty("registrationTime")
    public String registrationTime;

    @JsonProperty("completed")
    public boolean completed;

    @Nullable
    public ShootingTestAttemptSummary getAttemptSummaryFor(@Nullable final ShootingTestType type) {
        if (type != null) {
            for (final ShootingTestAttemptSummary item : attempts) {
                if (type == item.type) {
                    return item;
                }
            }
        }
        return null;
    }
}

File: riistaandroid/src/main/java/fi/riista/mobile/models/shootingTest/ShootingTestParticipant.kt
package fi.riista.mobile.models.shootingTest

import com.fasterxml.jackson.annotation.JsonProperty
import fi.riista.common.domain.model.ShootingTestType
import java.io.Serializable

class ShootingTestParticipant : Serializable {
    @JsonProperty("id")
    var id: Long = 0

    @JsonProperty("rev")
    var rev = 0

    @JsonProperty("firstName")
    var firstName: String? = null

    @JsonProperty("lastName")
    var lastName: String? = null

    @JsonProperty("hunterNumber")
    var hunterNumber: String? = null

    @JsonProperty("mooseTestIntended")
    var mooseTestIntended = false

    @JsonProperty("bearTestIntended")
    var bearTestIntended = false

    @JsonProperty("deerTestIntended")
    var deerTestIntended = false

    @JsonProperty("bowTestIntended")
    var bowTestIntended = false

    @JsonProperty("attempts")
    var attempts: List<ShootingTestAttemptSummary> = ArrayList()

    @JsonProperty("totalDueAmount")
    var totalDueAmount = 0

    @JsonProperty("paidAmount")
    var paidAmount = 0

    @JsonProperty("remainingAmount")
    var remainingAmount = 0

    @JsonProperty("registrationTime")
    var registrationTime: String? = null

    @JsonProperty("completed")
    var completed = false

    fun getAttemptSummaryFor(type: ShootingTestType?): ShootingTestAttemptSummary? {
        if (type != null) {
            return attempts.firstOrNull { it.type === type }
        }
        return null
    }
}
